src/zenml/steps/utils.py

```diff
@@ -393,7 +393,7 @@ def Do(
                 function_params[arg] = config_object
             elif issubclass(arg_type, StepContext):
                 output_artifacts = {k: v[0] for k, v in output_dict.items()}
-                context = StepContext(
+                context = arg_type(
```

Comment:
For my information, why did this need to change?

Reply:
This change is needed to cover the case where the step function signature specifies a subclass of `StepContext` instead, as is the case with `WhylogsStepContext`.

Reply:
Ah makes sense!